### PR TITLE
fixed some issues

### PR DESCRIPTION
**Pull request by Francy-chan**
* fixed crash with non-vanilla fluids when on dedicated server
* fixed unable to set up filters on dedicated server
* fixed basins containing the wrong amount of fluid

### CODE REVIEW

File: src/main/java/com/simibubi/create/events/CommonEvents.java [hunk 1]
```diff
@@ -252,6 +253,7 @@ public static void register() {
 		UseBlockCallback.EVENT.register(FluidBottleItemHook::preventWaterBottlesFromCreatesFluids);
 		ProjectileImpactCallback.EVENT.register(BlazeBurnerHandler::onThrowableImpact);
 		EntityReadExtraDataCallback.EVENT.register(ExtendoGripItem::addReachToJoiningPlayersHoldingExtendo);
+		UseBlockCallback.EVENT.register(FilteringHandler::onBlockActivated);
```
Comment (TropheusJ) [inline src/main/java/com/simibubi/create/events/CommonEvents.java:256]: what do these changes solve? I haven't noticed any issues with filters.

File: src/main/java/com/simibubi/create/foundation/tileEntity/behaviour/filtering/FilteringHandler.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ else if (ItemHandlerHelper.canItemStacksStack(toApply, filter))
 			Component formattedText = new TranslatableComponent(translationKey);
 			player.displayClientMessage(Lang.createTranslationTextComponent("logistics.filter." + feedback, formattedText)
 				.withStyle(ChatFormatting.WHITE), true);
-		}
+		//}
```
Comment (TropheusJ) [inline src/main/java/com/simibubi/create/foundation/tileEntity/behaviour/filtering/FilteringHandler.java:98]: this screams broken - see above.

File: src/main/java/com/simibubi/create/content/contraptions/fluids/actors/FillingRecipe.java [hunk 3]
```diff
@@ -68,13 +74,27 @@ public void addAssemblyFluidIngredients(List<FluidIngredient> list) {
...
+		} else if (fluid == Fluids.LAVA) {
+			translationKey = "block.minecraft.lava";
+		} else {
+			ResourceLocation id = Registry.FLUID.getKey(fluid);
+			String key = Util.makeDescriptionId("block", id);
+			String translated = I18n.get(key);
+			translationKey = translated.equals(key) ? Util.makeDescriptionId("fluid", id) : key;
+		}
+
+		return Lang.translate("recipe.assembly.spout_filling_fluid", new TranslatableComponent(translationKey).getString());
```
Comment (TropheusJ) [inline src/main/java/com/simibubi/create/content/contraptions/fluids/actors/FillingRecipe.java:80]: I'd like this code de-duplicated and moved into a util class, probably FluidUtil

File: src/main/java/com/simibubi/create/content/contraptions/goggles/IHaveGoggleInformation.java [hunk 4]
```diff
@@ -57,7 +62,21 @@ default boolean containedFluidTooltip(List<Component> tooltip, boolean isPlayerS
...
+				translationKey = "block.minecraft.water";
+			} else if (data.fluid() == Fluids.LAVA) {
+				translationKey = "block.minecraft.lava";
+			} else {
+				ResourceLocation id = Registry.FLUID.getKey(data.fluid());
+				String key = Util.makeDescriptionId("block", id);
+				String translated = I18n.get(key);
+				translationKey = translated.equals(key) ? Util.makeDescriptionId("fluid", id) : key;
+			}
+
```
Comment (TropheusJ) [inline src/main/java/com/simibubi/create/content/contraptions/goggles/IHaveGoggleInformation.java]: see above

**Review by TropheusJ — CHANGES_REQUESTED** [review]

File: src/main/java/com/simibubi/create/events/CommonEvents.java (hunk 1 above)
Comment (Francy-chan) [reply src/main/java/com/simibubi/create/events/CommonEvents.java:256]: this change only takes effect into a dedicated server (not a LAN world / integrated server)
when the use event being set only on client, the server doesnt know that you set up an filter, resulting in no changes at all

File: src/main/java/com/simibubi/create/foundation/tileEntity/behaviour/filtering/FilteringHandler.java (hunk 2 above)
Comment (Francy-chan) [reply src/main/java/com/simibubi/create/foundation/tileEntity/behaviour/filtering/FilteringHandler.java:98]: this allows the filter feedback text and the use action to be sent no matter if its a integrated server or dedicated server

File: src/main/java/com/simibubi/create/content/contraptions/fluids/actors/FillingRecipe.java (hunk 3 above)
Comment (Francy-chan) [reply src/main/java/com/simibubi/create/content/contraptions/fluids/actors/FillingRecipe.java:80]: yeah, I agree with you, just didnt knew where to put it, so a general, generic, approach was using duplicated code

**Comment by TropheusJ** [comment]
going to merge this since it's mostly fine, but I'm going to work on FilteringHandler::onBlockActivated more.

**Pull request merged**